addons/mail/models/models.py

```diff
@@ -31,3 +34,164 @@ def _notify_email_header_dict(self):
+                _('failed emails due to invalid email address'),
+                _(
+                    "%(failed_ignored_emails_counter)d email(s) failed / ignored to avoid blocking delivery to other recipients.\n"
+                    "Please make sure that recipient's email address really exists and if it's actually valid.\n"
+                    "You can check your mail notifications for more details about failed emails "
+                    "(an error message should help you diagnose the issue.)"
+                ) % {
+                    'failed_ignored_emails_counter': failed_ignored_emails_counter,
+                },
+                {
+                   "related_view_url": "/web#action=%(mail_sms_notifications_window_action)s&amp;view_type=list" % {
+                       'mail_sms_notifications_window_action': mail_sms_notifications_window_action
+                   },
+                   "related_view_text": "go to mail notifications"
+                }
+            ),
+
+            "iap__low_credits": lambda account_credits: (
+                _('Your credits balance is too low'),
+                _(
+                    "Your current balance reached %(account_credits)1.2f, It’s time to recharge your credits!\n"
+                    "To avoid service interruption (SMS, snailmail, lead enrichment...),"
+                    "please go to your 'IAP Portal' [Settings app > Odoo IAP > View my Services].\n"
+                    "From there, you can view your current balance, recharge your credits, "
+                    "review your consumption and set another reminder (by email) to when credits are low."
+                ) % {
+                    'account_credits': account_credits
+                },
+                {
+                   "related_doc": 'https://www.odoo.com/documentation/user/online/general/in_app_purchase/in_app_purchase.html'
+                }
+            ),
+
+            "base__cron": lambda record_link, cron_exception, cron_error_datetime, cron_database: (
+                _('Scheduled action failure'),
+                _(
+                    "An error occurred while running a cron job in Odoo.\n\n"
+                    "Failure details :\n"
+                    "- Execption: %(cron_exception)s\n"
+                    "- Occurred on: %(cron_error_datetime)s\n"
+                    "- Database: %(cron_database)s"
+                ) % {
+                    'cron_exception': cron_exception,
+                    'cron_error_datetime': cron_error_datetime,
+                    'cron_database': cron_database
+                },
+                {
+                    "related_view_url": record_link.get('url', False),
+                    "related_view_text": record_link.get('text', False),
+                }
+            )
+        }
+        # This way we can add messages in other models or even override genesis ones
+        if module_messages:
+            messages.update(module_messages)
```

Comment:
for me, this is not really relevant to authorise the caller to give what you expect to receive.
Better override the method to add the failure type (by extracting the message variable in another method).
like : 
```
def _get_admin_notification(self, failure_type):
    return _get_admin_failure_message()[failure_type]

def _get_admin_failure_message():
    messages = {
        "pif": "some message",
        "paf": "some other message"
    }
    return messages
```
Then you override _get_admin_failure_message() to add "pouf" failure type in messages :D

Reply:
If I understand correctly your idea, This is already implemented.

Actually what you see here is just a simplified version of '_get_admin_notification' (I removed the override here and added directly the message), but if you check [the example from google calendar sync](https://github.com/odoo/odoo/blob/74db403aa749a35516ad5f598c8ae0925d6f1a74/addons/google_calendar/models/res_users.py#L107) you'll see that '_get_admin_notification' is overrided to add new messages as you suggested :D

The lambdas were used to add params on '_notify_admins()' call ([see example from fetchmail](https://github.com/odoo/odoo/blob/81addb9a020f813337ee22d477c09c0313593197/addons/fetchmail/models/fetchmail.py#L211)) where 'failed' variable used in message when computed.